StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -263,6 +272,30 @@ class EmbeddedPaymentMethodsView: UIView {
             // Update instance states
             self.savedPaymentMethodButton = nil
         }
+        
+        // Update text on card row to say just "Card" instead of "New card" if needed
+        if let oldCardButton = selectionButtonMapping[.new(paymentMethodType: .stripe(.card))],
```

Comment:
Sorry, I'm not understanding when we should show "new card" vs "card". It's confusing because assuming a user has two payment methods (1) bank account and (1) card, switching between the payment methods changes "card" -> "new card" , and it looks like a bug-- but i'm guessing this is expected behavior?

1. Can we sync offline about this?
2. If this is really a requirement, we should add some UI tests to test for this.

Reply:
> Sorry, I'm not understanding when we should show "new card" vs "card"

Whenever the selected saved payment method (if any) is a card we should display "New card", otherwise just "Card". Note: In this context "selected saved payment method" refers to what the saved PM row is displaying it may not actually be "selected" in the embedded view.

>  switching between the payment methods changes "card" -> "new card" , and it looks like a bug-- but i'm guessing this is expected behavior

Yep this is expected behavior, the previous behavior was a bug of sorts, check vertical mode for expected behavior. The current behavior on master is actually somewhat inconsistent, the copy on the card button depends entirely on the default saved payment method at load time. So if bank is selected by default it is just "Card", but if card is selected by default it is "New card". But this copy never updates depending on the user interactions.

> If this is really a requirement, we should add some UI tests to test for this.

Yeah, this behavior should match vertical mode, so it makes sense to me that it is a requirement and was called out in the quality review.

re tests: We did update a [UI test](https://github.com/stripe/stripe-ios/pull/4300/files#diff-1528ed9f3c25572a57550dea4569a6e9e75a440611c0d09d316ba1f4fc7101e8R236) for this change, and many of our UI tests depend on the text on this button. The updated UI test exercises the exact flow that was called out in quality review. Then we have snapshot tests in `EmbeddedPaymentMethodsViewSnapshotTests`. I think that's probably enough tests for the slight copy difference. I don't think it warrants a dedicated UI test above what we already have IMO.


Happy to sync offline if needed!